### PR TITLE
Add quirk for swapping headphone and router device.

### PR DESCRIPTION
**Pull request by jusa**
When swap_headphone_speaker quirk is enabled use
AUDIO_DEVICE_OUT_SPEAKER for output-wired_headphone and AUDIO_DEVICE_OUT_WIRED_HEADPHONE for output-speaker.

### CODE REVIEW

File: src/common/droid-util.c [hunk 1]
```diff
@@ -122,6 +123,8 @@ static int input_stream_set_route(pa_droid_hw_module *hw_module, pa_droid_stream
 static int droid_set_parameters(pa_droid_hw_module *hw, const char *parameters);
 static bool droid_set_audio_source(pa_droid_hw_module *hw_module, audio_source_t audio_source);
 
+static pa_droid_hw_module *hw_primary;
```
Comment (spiiroin) [inline src/common/droid-util.c]: obscure portability nit, feel free to ignore: uninitialized -> bss -> zero initialized -> (strictly speaking) != null initialized

File: src/common/droid-util.c [hunk 2]
```diff
@@ -445,15 +448,22 @@ static void add_o_ports(pa_droid_mapping *am) {
...
         uint32_t cur_device = (1 << i++);
 
         if (devices & cur_device) {
 
-            pa_assert_se(pa_droid_output_port_name(cur_device, &name));
+            if (pa_droid_quirk(hw_primary, QUIRK_SWAP_HEADPHONE_SPEAKER)) {
+                if (cur_device & AUDIO_DEVICE_OUT_WIRED_HEADPHONE)
+                    pa_assert_se(pa_droid_output_port_name(AUDIO_DEVICE_OUT_SPEAKER, &name));
+                else if (cur_device & AUDIO_DEVICE_OUT_SPEAKER)
+                    pa_assert_se(pa_droid_output_port_name(AUDIO_DEVICE_OUT_WIRED_HEADPHONE, &name));
```
Comment (spiiroin) [inline src/common/droid-util.c:465]: what if it is neither speaker nor headphone - i.e. missing an 'else' block?

File: src/common/droid-util.c [hunk 3]
```diff
@@ -1062,6 +1072,8 @@ static pa_droid_hw_module *droid_hw_module_open(pa_core *core, const pa_droid_co
         pa_shared_set(core, DROID_HW_HANDLE_V1, hw);
         pa_shared_set(core, DROID_SET_PARAMETERS_V1, droid_set_parameters_v1_cb);
         pa_shared_set(core, DROID_GET_PARAMETERS_V1, droid_get_parameters_v1_cb);
+
+        hw_primary = hw;
```
Comment (spiiroin) [inline src/common/droid-util.c]: any chance that those pa_shared_set() calls could now or in the future span somewhere where hw_primary is used - i.e. should the assignment be done 1st?

File: src/common/droid-util.c (hunk 2 above)
Comment (jusa) [reply src/common/droid-util.c:465]: Fixed.

File: src/common/droid-util.c (hunk 3 above)
Comment (jusa) [reply src/common/droid-util.c]: There's always a chance :) Changed order.

**Review by mlehtima — APPROVED** [review]

**Review by spiiroin — APPROVED** [review]

**Pull request merged**